### ISSUE TITLE
added a new yandex translation service

### ISSUE THREAD

**Issue by hrach333** [butaixianran/Stable-Diffusion-Webui-Prompt-Translator#11]
I have added a new yandex translation service, if necessary, I can write instructions for it

**Comment by hrach333** [comment]
![image](https://user-images.githubusercontent.com/698064/225091539-ab714be6-4916-475b-90a2-ec939e7fdec2.png)

**Comment by butaixianran** [comment]
I don't use yandex so I can not test it. 

But if you like, you can create a pull request. Also, you need to update the document too, I don't use yandex so I can not write the document for you.

**Comment by hrach333** [comment]
You didn't quite understand me correctly, I can do the documentation myself. And I made a change

**Comment by piketz** [comment]
a couple of weeks ago I already added Yandex api.  but its auto language detector doesn't work well.  if there are more than a certain number of words, then it does not translate from the main language.  I had to add the choice of the original language

**Comment by hrach333** [comment]
I have tried to translate in different languages, even in Chinese, everything is fine and translates.

**Comment by hrach333** [comment]
I also wrote a large promt for translation, he translated it without any problems
![image](https://user-images.githubusercontent.com/698064/225109733-c3d40eec-66c1-4fed-8b24-3bb236f60d21.png)

**Comment by hrach333** [comment]
I created a pull request

**Comment by piketz** [comment]
use a big prompt in English and add the last word in Russian.  he won't transfer it.

**Comment by hrach333** [comment]
> use a big prompt in English and add the last word in Russian. he won't transfer it.

yes, you are right, there is such a thing, but still it works, maybe there is a solution in the multilingual version. We need to see.

**Comment by piketz** [comment]
I solved this problem by using a model to translate the text.  and there is no need to use third-party services.  transfer speed is about the same.  for 2060s.

**Comment by hrach333** [comment]
what is the model?

**Comment by butaixianran** [comment]
Like I mentioned in another issue, there is really no need to add a new feature just because you want to add one word with native language to the end of a English prompt.

You can just put this word into neg-prompt, translate it into English, then put it back to prompt. 

If you always add one word to English prompt with your native language in every cases, you can use  Style function to save this addtional word as template. Or you can use a similar extension, like this one:
https://github.com/butaixianran/sdweb-easy-prompt-selector

**Comment by piketz** [comment]
> what is the model?

check https://github.com/butaixianran/Stable-Diffusion-Webui-Prompt-Translator/issues/8
and my fork. https://github.com/piketz/Stable-Diffusion-Webui-Prompt-Translator

> You can just put this word into neg-prompt, translate it into English, then put it back to prompt.

It seems more convenient to translate the line line
"En, en, en, en, en, _ru_, en, en, en, en, en, en, en,  _ru_, _ru_, en, en"

**Comment by butaixianran** [comment]
hi, I merged your pull request, but it only updates scripts, not this project 's README.md Document.

As I mentioned above, You need to add a new section into readme.md as tutorial for how to use yandex to get APP Key.

Without document, other users of this extension won't be able to use yandex even this feature is already in code.

**Comment by butaixianran** [comment]
Also, run git pull to get latest version, before you update Readme.md document. I may do some update recently.

**Comment by butaixianran** [comment]
closed since no reply

**Comment by hrach333** [comment]
sorry, I just didn't see it right away, I'll write a readme in the instructions

**Comment by sebaxakerhtc** [comment]
> sorry, I just didn't see it right away, I'll write a readme in the instructions

Ёу!
You should add an instructions because i have killed a 1 hour to understand on "how to"...
And especially @butaixianran, please rename `APP ID` to `Folder ID` and `API key` to `IAM token` from [instruction](https://cloud.yandex.com/en/docs/translate/operations/translate)

**Comment by sebaxakerhtc** [comment]
> > sorry, I just didn't see it right away, I'll write a readme in the instructions
> 
> Ёу! You should add an instructions because i have killed a 1 hour to understand on "how to"... And especially @butaixianran, please rename `APP ID` to `Folder ID` and `API key` to `IAM token` from [instruction](https://cloud.yandex.com/en/docs/translate/operations/translate)

In this part
```
        gr.Markdown("Translation Service Setting")
        provider = gr.Dropdown(choices=providers, value=provider_name, label="Provider", elem_id="pt_provider")
        app_id = gr.Textbox(label="APP ID", lines=1, value=trans_setting[provider_name]["app_id"], elem_id="pt_app_id")
        app_key = gr.Textbox(label="APP KEY", lines=1, value=trans_setting[provider_name]["app_key"], elem_id="pt_app_key")
        save_trans_setting_btn = gr.Button(value="Save Setting")
```
Something like `if provider_name == "Yandex":` then `app_id.label = "Folder Id", app_key.label = "IAM token"`
Sorry for errors - i'm not a Python dev, so i can't write a perfect code, but I think you can understand what i mean...
@butaixianran

**Comment by butaixianran** [comment]
label name won't be changed. There should be a notice in the yandex section of document to tell users, fill Folder ID into App ID if you are using yandex.

Again, I don't use yandex, I can not write this document for you. If you know how, you can write that part of the document and create a pull request.

**Comment by butaixianran** [comment]
Document is updated

**Comment by sebaxakerhtc** [comment]
> label name won't be changed

I'm sure i'll find better way, but for now i make just cosmetic changes...
```
        gr.Markdown("Translation Service Setting")
        provider = gr.Dropdown(choices=providers, value=provider_name, label="Provider", elem_id="pt_provider")
        app_id = gr.Textbox(label="APP ID / Folder ID", lines=1, value=trans_setting[provider_name]["app_id"], elem_id="pt_app_id")
        app_key = gr.Textbox(label="APP KEY / IAM Token", lines=1, value=trans_setting[provider_name]["app_key"], elem_id="pt_app_key")
        save_trans_setting_btn = gr.Button(value="Save Setting")
```

![1](https://user-images.githubusercontent.com/32651506/229216963-e48a97a4-a496-46b3-95e4-3f96dfdc0528.png)

**Comment by butaixianran** [comment]
Most people won't choose yandex in this extension. They will be confused what is folder id and IAM token.  

It is yandex who doesn't follow industry naming standard. And document is already mentioned this, it is good enough.
